### PR TITLE
Automated cherry pick of #85610: Bump Cluster Autoscaler version to 1.17.0

### PR DESCRIPTION
**Pull request by losipiuk**
Cherry pick of #85610 on release-1.17.

#85610: Bump Cluster Autoscaler version to 1.17.0

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

### CODE REVIEW

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve

**Comment by justaugustus** [comment]
Will lift hold once the autoscaling e2e passes:
/hold
/assign

**Comment by justaugustus** [comment]
Release Manager approval:
/lgtm
/approve
/hold cancel

**Pull request merged**
